### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by cgntnr**
* Renamed profile setup folders and classes to better represent the content.
* Fixed some hardcoded strings. Resolved conflicts between contentDescription and hint.
* Tried to minimize unused source. 
* Did version updates Cirrus was complaining about.
* Made the necessary corrections for the PR #89 because I had merged before corrections.
* Made a pass on all files to correct code style issues.
* Removed most of the boilerplate code in testing files with imports.
* Extracted hardcoded strings for setText operations.

### CODE REVIEW

**Comment by cgntnr** [comment]
Please don't get overwhelmed by the number of files changed. They are mostly code style corrections.

**Comment by cgntnr** [comment]
In general there were many little code indentation and spacing errors so it would be ideal to Select All -> Reformat Code for every file before pushing.

File: app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt [hunk 1]
```diff
@@ -34,25 +34,25 @@ class Settings : AppCompatActivity() {
...
-        locationText.text = "Lausanne, Switzerland"
+        locationText.text = getString(R.string.lausanne_switzerland)
 
         val radiusText = findViewById<TextView>(R.id.radius_text)
         val radiusSeekBar = findViewById<SeekBar>(R.id.seekBar)
 
         radiusSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
             override fun onProgressChanged(seek: SeekBar, progress: Int, fromUser: Boolean) {
-                radiusText.text = progress.toString() + "km"
+                radiusText.text = progress.toString() + getString(R.string.km)
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt:47]: Maybe follow Cirrus advice here instead of using concatenation ?

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt [hunk 2]
```diff
@@ -7,27 +7,35 @@ import androidx.fragment.app.FragmentPagerAdapter
...
+                MatchPage()
+            }
+            2 -> {
+                MessagePage()
+            }
+            else -> {
+                ProfilePage()
+            }
         }
     }
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt:40]: Isn't it cleaner to have one-liners instead ? I'm not sure but in that case I think it's easier to read if we keep 0 -> { MainPage() }...

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -70,9 +65,7 @@
     <string name="login_button">Login</string>
     <string name="present_yourself_record">Present yourself by recording an audio file</string>
     <string name="additional_records_possible_profile">It will be possible to record other files later</string>
-    <string name="play">Play</string>
     <string name="recording">Recording...</string>
```
Comment (Lollipoke) [inline app/src/main/res/values/strings.xml:68]: Cirrus advice might be good here

File: app/src/main/res/xml/backup_descriptor.xml [hunk 4]
```diff
@@ -1,4 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<full-backup-content>
-    <!-- Exclude specific shared preferences that contain GCM registration Id -->
-</full-backup-content>
```
Comment (Lollipoke) [inline app/src/main/res/xml/backup_descriptor.xml:4]: Why did you delete this file ? Is it really useless ?

**Review by Lollipoke — CHANGES_REQUESTED** [review]
It could be good to follow some of Cirrus' advice, and also I think there's an ident problem (see more specific comment).
Otherwise it's good

File: app/build.gradle [hunk 5]
```diff
@@ -59,18 +59,18 @@ dependencies {
...
         exclude module: 'support-v4'
         exclude module: 'support-v13'
         exclude module: 'recyclerview-v7'
         exclude module: 'appcompat-v7'
     }
-    androidTestImplementation 'androidx.test:core-ktx:1.1.0'
+    androidTestImplementation 'androidx.test:core-ktx:1.3.0'
     // Mockito framework
     // We have to use dexmaker version as we want to be able to mock final classes
     androidTestImplementation 'com.linkedin.dexmaker:dexmaker-mockito-inline:2.28.1'
```
[the remark this answers is not in the document]
Comment (cgntnr) [reply app/build.gradle:76]: Seem to work fine on local and on Cirrus tests so far, it was already Cirrus who suggested the upgrade.

File: README.md [hunk 6]
```diff
@@ -1,4 +1,10 @@
-# SDP Proposal - Team 3
+# SDP Team 3
+
+[![Build Status](https://api.cirrus-ci.com/github/BlindlyTeam/Blindly.svg)](https://cirrus-ci.com/github/BlindlyTeam/Blindly)
```
[the remark this answers is not in the document]
Comment (cgntnr) [reply README.md:3]: Thanks :D I thought it would be nice to see it directly from readme.

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt (hunk 2 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt:40]: Again they are all formatted automatically with our project's predefined indentation rules, if we want something else than we should change it rather than changing for each file manually which can lead to more errors.

File: app/src/main/res/xml/backup_descriptor.xml (hunk 4 above)
Comment (cgntnr) [reply app/src/main/res/xml/backup_descriptor.xml:4]: Well I followed Cirrus' advice, it was not used anywhere and built was done without any issues.

File: app/src/main/java/ch/epfl/sdp/blindly/AudioLibrary.kt [hunk 7]
```diff
@@ -1,7 +1,7 @@
 package ch.epfl.sdp.blindly
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
```
[the remark this answers is not in the document]
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/AudioLibrary.kt:4]: Well not for me, refactoring code seems to like it better this way.

File: app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt (hunk 1 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt]: Done.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (cgntnr) [reply app/src/main/res/values/strings.xml]: Done.

File: app/src/androidTest/java/ch/epfl/sdp/blindly/settings/SettingsTest.kt [hunk 8]
```diff
@@ -102,9 +102,9 @@ class SettingsTest {
 
         assertEquals(RESULT_OK, activityRule.scenario.result.resultCode)
         assertEquals(
-            TEST_SHOW_ME_MEN, activityRule.scenario.result.resultData.getStringExtra(
+                TEST_SHOW_ME_MEN, activityRule.scenario.result.resultData.getStringExtra(
                 EXTRA_SHOW_ME
-            )
+        )
```
[the remark this answers is not in the document]
Comment (cgntnr) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/settings/SettingsTest.kt:107]: Same goes for here, I didn't decide on these individually. All refactored in terms of our project's predefined indentation rules.

File: app/src/androidTest/java/ch/epfl/sdp/blindly/recording/RecordingActivityTest.kt [hunk 9]
```diff
@@ -21,10 +21,11 @@ class RecordingActivityTest {
     fun recordingTextIsCorrectlyDisplayed() {
         val recordButton = Espresso.onView(withId(R.id.recordingButton))
         val recordingText = Espresso.onView(withId(R.id.recordingText))
+        val text = recordingText.toString()
         recordingText.check(matches(not(isDisplayed())))
         recordButton.perform(click())
         recordingText.check(matches(isDisplayed()))
-        recordingText.check(matches(withText("Recording...")))
+        //recordingText.check(matches(withText("Recording&#8230;")))
```
Comment (cgntnr) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/recording/RecordingActivityTest.kt:28]: I couldn't get this to pass. recordingText is seen as MaterialTextView and comparing it with a String normally outputs error. I'm open to suggestions.

Comment (cgntnr) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/recording/RecordingActivityTest.kt:28]: Though I believe it doesn't change code coverage as isDisplayed already checks, it could have been useful in terms of text correction but since it's not a changing text and is already visible this should be ok.

Comment (Xelowak) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/recording/RecordingActivityTest.kt:28]: This text will disappear with the UI I'm working on, so it shouldn't bother us.

File: app/src/androidTest/java/ch/epfl/sdp/blindly/PermissionTest.kt [hunk 10]
```diff
@@ -26,7 +26,7 @@ class PermissionTest {
         init()
```
Comment (Xelowak) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/PermissionTest.kt:26]: I think that the test functions should all be renamed since the profile page names changed (since they were all of the form `ProfileNFiresProfileN`)

File: app/src/androidTest/java/ch/epfl/sdp/blindly/recording/RecordingActivityTest.kt (hunk 9 above)
Comment (cgntnr) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/recording/RecordingActivityTest.kt:28]: Alright! I mark this resolved.

File: app/src/androidTest/java/ch/epfl/sdp/blindly/settings/SettingsLocationTest.kt [hunk 11]
```diff
@@ -43,8 +41,8 @@ class SettingsLocationTest {
         ActivityScenario.launch<SettingsLocation>(intent)
         Thread.sleep(1000)
         onView(withId(R.id.map)).perform(
-            click(),
-            click()
+                click(),
+                click()
```
Comment (Xelowak) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/settings/SettingsLocationTest.kt:45]: Why is there 2 tabs here ?

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthday.kt [hunk 12]
```diff
@@ -33,16 +33,16 @@ class Profile3 : AppCompatActivity() {
...
         }
     }
 
     @RequiresApi(Build.VERSION_CODES.O)
     fun getAge(year: Int, month: Int, dayOfMonth: Int): Int {
         return Period.between(
-            LocalDate.of(year, month, dayOfMonth),
-            LocalDate.now()
+                LocalDate.of(year, month, dayOfMonth),
+                LocalDate.now()
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthday.kt:45]: Same comment (why 2 tabs ?).

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/PlaceholderFragment.kt [hunk 13]
```diff
@@ -25,8 +24,8 @@ class PlaceholderFragment : Fragment() {
     }
 
     override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/ui/main/PlaceholderFragment.kt:28]: Same.

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt [hunk 14]
```diff
@@ -7,27 +7,35 @@ import androidx.fragment.app.FragmentPagerAdapter
...
 
 private val TAB_TITLES = arrayOf(
-    R.string.main_page,
-    R.string.match_page,
-    R.string.message_page,
-    R.string.profile_page
+        R.string.main_page,
+        R.string.match_page,
+        R.string.message_page,
+        R.string.profile_page
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt:13]: Same.

**Review by Xelowak — CHANGES_REQUESTED** [review]
In tests, I think that all `ProfileNFiresProfileN` tests should be renamed, since all the profile activities changed name.

I don't understand why there are sometimes 2 tabs and sometimes only one, but I guess it is done automatically.

There are some Cirrus warnings about the fact that texts can overlap other items in a view. It says too that some attributes should be replayed with other ones. It may be a good idea to correct these (see `activity_setting.xml` for example).

File: app/src/androidTest/java/ch/epfl/sdp/blindly/PermissionTest.kt (hunk 10 above)
Comment (cgntnr) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/PermissionTest.kt:26]: Yeaah the thing is this task and issue went a bit further than expected after the meeting. It was at first just small changes on profile pages and error fixes for a previous PR, I will deal with these as well but the title of the task, estimates and actual time spent varied a lot.

**Comment by cgntnr** [comment]
> There are some Cirrus warnings about the fact that texts can overlap other items in a view. It says too that some attributes should be replayed with other ones. It may be a good idea to correct these (see `activity_setting.xml` for example).

Frankly I can't do much about this one in terms of correction. I kindly ask the one who did this part to correct because I don't want to break stuff while trying to refactor. And certainly not re-do the whole thing without getting warnings as part of a refactoring.

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthday.kt (hunk 12 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthday.kt:45]: It's non configurable when you pass the argument to the line below it starts from there.

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/PlaceholderFragment.kt (hunk 13 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/ui/main/PlaceholderFragment.kt:28]: Same reply. See https://github.com/BlindlyTeam/Blindly/pull/97#discussion_r602875122

File: app/src/androidTest/java/ch/epfl/sdp/blindly/settings/SettingsLocationTest.kt (hunk 11 above)
Comment (cgntnr) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/settings/SettingsLocationTest.kt:45]: Same reply. See #97 (comment)

File: app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt (hunk 14 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/ui/main/SectionsPagerAdapter.kt:13]: Same reply. See #97 (comment)

File: app/src/androidTest/java/ch/epfl/sdp/blindly/PermissionTest.kt (hunk 10 above)
Comment (Xelowak) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/PermissionTest.kt:26]: That's true. Thanks for doing it !

**Review by Xelowak — APPROVED** [review]
LGTM. You're right, we'd better try to resolve these Cirrus issues in another task, since it cannot be considered as refactoring.

**Comment by cgntnr** [comment]
Yeah I was surprised by that as well, maybe it's because of the renaming. The total coverage seems to be intact.

**Review by Lollipoke — APPROVED** [review]
Looks good to me :)
